gems/smithy/lib/smithy/anvil/client/templates/endpoint_parameters.erb

```diff
@@ -0,0 +1,64 @@
+# frozen_string_literal: true
+
+# This is generated code!
+
+module <%= namespace %>
+  # Endpoint parameters used to influence endpoints per request.
+  <%= documentation %>
+  EndpointParameters = Struct.new(
+    <%- parameters.each do |param| -%>
+    :<%= param.name %>,
+    <%- end -%>
+    keyword_init: true
+  ) do
+    include Smithy::Client::Structure
+
+    # @api private
+    class << self
+      PARAM_MAP = {
+      <%- parameters.each do |param| -%>
+        "<%= param.id %>" => :<%= param.name %>,
+      <%- end -%>
+      }.freeze
+    end
+
+    def initialize(options = {})
+    <%- parameters.each do |param| -%>
+      self[:<%= param.name %>] = options.fetch(:<%= param.name %>, <%= param.default_value %>)
+    <%- end -%>
+    <%- parameters.select(&:validate_required?).each do |param| -%>
+      raise ArgumentError, "Missing required EndpointParameter: :<%= param.name %>" if <%= param.name %>.nil?
+    <%- end -%>
+    end
+
+    def self.create(config, operation_name=nil, params={})
+      <%- if operation_params.empty? -%>
+      new({
+      <%- parameters.select(&:create_from_config?).each do |param| -%>
+        <%= param.name %>: <%= param.value %>,
+      <%- end -%>
+      }.compact)
```

Comment:
Is compact necessary? You are selecting values that should not be nil.

Reply:
They can be nil here and if so, they should get default values (if applicable) so we do need the compact.

Reply:
Could we just ignore nil values in our select?

Reply:
I dont think so - we don't know at codegen time what runtime values users are going to use?

Reply:
Makes sense.